src/test/java/org/rulez/demokracia/pdengine/VoteTest.java

```diff
@@ -25,13 +25,28 @@ public void setUp() throws ReportedException {
 	@tested_operation("Cast vote")
 	@tested_behaviour("deletes the ballot with ballotId, so only one vote is possible with a ballot")
 	@Test
-	public void cast_vote_deletes_our_ballot() {
+	public void cast_vote_deletes_ballot_if_canVote_is_true() {
 		String ballot = voteManager.obtainBallot(adminInfo.voteId, adminInfo.adminKey);
 		List<RankedChoice> theCastVote = new ArrayList<RankedChoice>();
 		Vote vote = getTheVote();
+		vote.canVote = true;
 		voteManager.castVote(adminInfo.voteId,ballot,theCastVote);
-		assertFalse(vote.ballots.contains(ballot));
-		
+		assertFalse(vote.ballots.contains(ballot));		
```

Comment:
I do not believe this would get executed.

Reply:
Why? The ballot will delete from the ballots list. So the ballots will not contain the ballot. Thatswhy I use assertFalse.

Reply:
Oops, my bad, I was thinking it is a canVote=false case.

Reply:
My bad. Thought it is a canVote=false case.